internal/pkg/docker/dockerfile/dockerfile.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+package dockerfile
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+const exposeRegexPattern = `(\d+)(\/(\w+))?` // port and optional protocol, at least 1 time on a line
+
+var (
+	errCouldntParseDockerfilePort = errors.New("parse port from EXPOSE")
+)
+
+type Dockerfile interface {
+	GetExposedPorts() []uint16
+}
+
+type PortConfig struct {
+	Port      uint16
+	Protocol  string
+	RawString string
+}
+
+type Config struct {
+	ExposedPorts []PortConfig
+	parsed       bool
+	path         string
+
+	fs afero.Fs
+}
+
+func NewConfig(fs afero.Fs, path string) *Config {
+	return &Config{
+		ExposedPorts: []PortConfig{},
+		fs:           fs,
+		path:         path,
+		parsed:       false,
+	}
+}
+
+func (df *Config) GetExposedPorts() []uint16 {
+	if !df.parsed {
+		df.parse()
+	}
+
+	ports := []uint16{}
+	for _, port := range df.ExposedPorts {
+		ports = append(ports, port.Port)
+	}
+	return ports
+}
+
+// ParseDockerfile takes a Dockerfile and struct of methods and returns a json representation
```

Comment:
It seems like the function name in the comment does not comply with the function name.